staging/src/k8s.io/apimachinery/pkg/util/yaml/decoder.go

```diff
@@ -343,6 +343,12 @@ func GuessJSONStream(r io.Reader, size int) (io.Reader, []byte, bool) {
 	return buffer, b, hasJSONPrefix(b)
 }
 
+// IsJSONBuffer scans the provided buffer, looking
```

Comment:
this is no more correct than `GuessJSONStream` and still fails https://github.com/kubernetes/kubernetes/issues/78946

Reply:
I'm not in any way addressing those other bugs.  This change isn't fixing known gaps with guess.

Reply:
(i'm all for improving guess as efficiently as possible, which means finding all callers and having them all consistently use a bigger buffer, but there is no solution other than reading it all into memory to get perfection)